pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -342,13 +342,21 @@ def on_save_checkpoint(
             "best_model_path": self.best_model_path,
             "current_score": self.current_score,
             "dirpath": self.dirpath,
+            "best_k_models": self.best_k_models,
+            "kth_best_model_path": self.kth_best_model_path,
+            "kth_value": self.kth_value,
+            "last_model_path": self.last_model_path,
         }
 
     def on_load_checkpoint(
         self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", callback_state: Dict[str, Any]
     ) -> None:
         self.best_model_score = callback_state["best_model_score"]
         self.best_model_path = callback_state["best_model_path"]
+        self.best_k_models = callback_state.get("best_k_models", self.best_k_models)
```

Comment:
why `.get`?
If those are saved. They should be available?

Reply:
Yes, but if you don't use get, this won't work with legacy checkpoints :)

Reply:
good point 👍 
do we have an interval or some sort of minimal version wrt to the current pl version until which we support legacy checkpoints?

Reply:
We test for all versions since 1.0.0